packages/core/src/components/time-picker/time-picker.tsx

```diff
@@ -131,43 +202,248 @@ export class TimePicker {
       return;
     }
 
-    this._timeRef = this.format.includes('a')
-      ? (DateTime.fromFormat(this.time, this.format).toFormat('a') as
-          | 'AM'
-          | 'PM')
-      : undefined;
-    this.formatTime();
+    this.setTimeRef();
+    this.focusedValue =
+      this._time.hour ??
+      this._time.minute ??
+      this._time.second ??
+      this._time.millisecond;
+    this.formattedTime = this.getFormattedTime();
+    this.setTimePickerDescriptors();
   }
 
-  @Watch('_time')
-  formatTime() {
-    if (!this._time) {
+  componentDidLoad() {
+    this.updateScrollPositions();
+    this.setupVisibilityObserver();
+  }
+
+  disconnectedCallback() {
+    if (this.visibilityObserver) {
+      this.visibilityObserver.disconnect();
+      this.visibilityObserver = undefined;
+    }
+  }
+
+  @OnListener<TimePicker>('keydown')
+  handleKeyDown(event: KeyboardEvent) {
+    if (!this.isUnitFocused && !this.ampmFocus) {
       return;
     }
 
-    const [hour, minute, second] = this._time
-      .toFormat(this.format)
-      .split(' ')[0]
-      .split(':');
+    let newValue = this.focusedValue;
+    let shouldPreventDefault = true;
+    const newValueInterval =
+      this.focusedUnit === 'hour'
+        ? this.hourInterval
+        : this.focusedUnit === 'minute'
+        ? this.minuteInterval
+        : this.focusedUnit === 'second'
+        ? this.secondInterval
+        : this.millisecondInterval;
```

Comment:
Bad readability can we resolve the newValueInterval without nested ternary operator?